google/cloud/aiplatform/_featurestores/entity_type.py

```diff
@@ -0,0 +1,456 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from typing import Dict, List, Optional, Sequence, Tuple, Union
+
+import datetime
+import pandas as pd
+
+from google.auth import credentials as auth_credentials
+from google.protobuf import field_mask_pb2
+
+from google.cloud.aiplatform import base
+from google.cloud.aiplatform.compat.types import entity_type as gca_entity_type
+from google.cloud.aiplatform import _featurestores
+from google.cloud.aiplatform import utils
+
+from google.cloud.aiplatform.utils import featurestore_utils
+
+_LOGGER = base.Logger(__name__)
+_ALL_FEATURE_IDS = "*"
+
+
+class EntityType(base.VertexAiResourceNounWithFutureManager):
+    """Managed entityType resource for Vertex AI."""
+
+    client_class = utils.FeaturestoreClientWithOverride
+
+    _is_client_prediction_client = False
+    _resource_noun = None
+    _getter_method = "get_entity_type"
+    _list_method = "list_entity_types"
+    _delete_method = "delete_entity_type"
+
+    def __init__(
+        self,
+        entity_type_name: str,
+        featurestore_id: Optional[str] = None,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        """Retrieves an existing managed entityType given an entityType resource name or an entity_type ID.
+
+        Example Usage:
+
+            my_entity_type = aiplatform.EntityType(
+                entity_type_name='projects/123/locations/us-central1/featurestores/my_featurestore_id/\
+                entityTypes/my_entity_type_id'
+            )
+            or
+            my_entity_type = aiplatform.EntityType(
+                entity_type_name='my_entity_type_id',
+                featurestore_id='my_featurestore_id',
+            )
+
+        Args:
+            entity_type_name (str):
+                Required. A fully-qualified entityType resource name or an entity_type ID.
+                Example: "projects/123/locations/us-central1/featurestores/my_featurestore_id/entityTypes/my_entity_type_id"
+                or "my_entity_type_id" when project and location are initialized or passed, with featurestore_id passed.
+            featurestore_id (str):
+                Optional. Featurestore to retrieve entityType from.
```

Comment:
What is the value if not set?

Reply:
The optional `featurestore_id` defaults to `None`, see couple of lines above. 
```
featurestore_id: Optional[str] = None,
```

Reply:
I mean if it's None, does it mean the value from the entity_type_name name will be used, which implies entity_type_name must be the fully qualified path. And when it's None, then entity_type_name must be the short ID form. Probably adding some explanation to be clear